src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -599,11 +595,11 @@ class JavaThread: public Thread {
   bool cont_fastpath() const                   { return _cont_fastpath == nullptr && _cont_fastpath_thread_state != 0; }
   bool cont_fastpath_thread_state() const      { return _cont_fastpath_thread_state != 0; }
 
-  void inc_held_monitor_count(int i = 1, bool jni = false);
-  void dec_held_monitor_count(int i = 1, bool jni = false);
+  void inc_held_monitor_count(intx i = 1, bool jni = false);
+  void dec_held_monitor_count(intx i = 1, bool jni = false);
```

Comment:
Does the parameter type need to be changed? AFAICS we never increment by other than one in any case (and inc/dec imply one - else they should be add/sub).

Reply:
Yes, the parameter type needed to be changed because:
```
  void do_monitor(ObjectMonitor* mid) {
    intx rec = mid->complete_exit(_thread);
    _thread->dec_held_monitor_count(rec + 1);
  }
```

In general, I only widen things when there's a warning produced.